File: lib/network/packets/client/c_map_info.dart
import '../../opcodes/client_opcodes.dart';

/// 請求當前地圖資訊（小地圖：地名 + 傳送點清單）。
/// 伺服器依角色當前 mapId 回 S_MAP_INFO；通常用於重整小地圖。
class CMapInfo {
  CMapInfo._();

  static Map<String, dynamic> build() => {
        'op': ClientOpcodes.cMapInfo,
        'data': <String, dynamic>{},
      };
}

File: src/main/java/com/xin/server/packet/client/C_MapInfo.java
package com.xin.server.packet.client;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.xin.server.model.instance.PcInstance;
import com.xin.server.network.Client;
import com.xin.server.packet.ClientBasePacket;
import com.xin.server.packet.server.S_MapInfo;

/**
 * 請求地圖資訊（小地圖：地名 + 傳送點清單）。
 * <p>
 * 客戶端 JSON 格式：
 * <pre>
 * { "op": "C_MAP_INFO", "data": {} }
 * </pre>
 *
 * 伺服器依角色當前 mapId 回傳 {@link S_MapInfo}。
 * 通常用於前端重新整理小地圖（如重新連線後同步）。
 */
public class C_MapInfo extends ClientBasePacket {

    private static final Logger logger = LoggerFactory.getLogger(C_MapInfo.class);

    public C_MapInfo(String raw) {
        super(raw);
    }

    @Override
    public void run(Client client) {
        if (!client.hasActiveChar()) {
            logger.warn("C_MAP_INFO 失敗：尚未進入遊戲");
            return;
        }
        PcInstance pc = client.getActiveChar();
        client.sendPacket(S_MapInfo.of(pc.getMapId()));
    }
}
